File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.*;
import java.util.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());
        ArrayDeque<Integer> q = new ArrayDeque<>();

        for (int i = 0; i < n; i++) {
            String operation = br.readLine();
            if (operation.contains("push")) {
                int pushNum = Integer.parseInt(operation.split(" ")[1]);

                q.add(pushNum);
                continue;
            }

            if (operation.equals("pop")) {
                if (q.isEmpty()) {
                    bw.write(-1 + "\n");
                } else {
                    bw.write(q.poll() + "\n");
                }
                continue;
            }

            if (operation.equals("size")) {
                bw.write(q.size() + "\n");
                continue;
            }

            if (operation.equals("empty")) {
                if (q.isEmpty()) {
                    bw.write(1 + "\n");
                } else {
                    bw.write(0 + "\n");
                }
                continue;
            }

            if (operation.equals("front")) {
                if (q.isEmpty()) {
                    bw.write(-1 + "\n");
                } else {
                    bw.write(q.peek() + "\n");
                }
                continue;
            }

            if (operation.equals("back")) {
                if (q.isEmpty()) {
                    bw.write(-1 + "\n");
                } else {
                    bw.write(q.getLast() + "\n");
                }
            }
        }

        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/18258. 큐 2/큐 2.py
from collections import deque
import sys

n = int(input())
q = deque()
for _ in range(n):
    s = sys.stdin.readline().rstrip()
    if 'push' in s:
        com, count = s.split(' ')
        q.append(count)
    elif s == 'front':
        if len(q) == 0:
            print(-1)
        else:
            print(q[0])
    elif s == 'back':
        if len(q) == 0:
            print(-1)
        else:
            print(q[-1])
    elif s == 'pop':
        if len(q) == 0:
            print(-1)
        else:
            print(q.popleft())
    elif s == 'size':
        print(len(q))
    elif s == 'empty':
        if len(q) == 0:
            print(1)
        else:
            print(0)